iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java

```diff
@@ -291,9 +291,17 @@ protected void updateCache(final Predicate<SwitchImpl> terminate) {
             CalculatedBus[] node2bus = new CalculatedBus[graph.getVertexCapacity()];
             boolean[] encountered = new boolean[graph.getVertexCapacity()];
             Arrays.fill(encountered, false);
-            for (int e : graph.getEdges()) {
-                traverse(graph.getEdgeVertex1(e), encountered, terminate, id2bus, node2bus);
-                traverse(graph.getEdgeVertex2(e), encountered, terminate, id2bus, node2bus);
+            if (graph.getEdgeCount() == 0) {
```

Comment:
@mathbagu can you please clean this function for me ? I am not sure I am able to. Thanks !